src/Serialization/polymake.jl

```diff
@@ -0,0 +1,153 @@
+    conv = Polymake.get_attachment(lp, "convention")
+    if isnothing(conv)
+        throw(ArgumentError("Incomplete or corrupt data file."))
+    end
+    if conv == "max"
+        return LinearProgram(Polyhedron(fr), lp, :max)
+    elseif conv == "min"
+        return LinearProgram(Polyhedron(fr), lp, :min)
+    end
+end
+
+
+
+
+
+
+##############################################################################
+"""
+    load_from_polymake(::Type{SubdivisionOfPoints}, jsondict::Dict{Symbol, Any})
+
+Load a subdivision of points stored in JSON format, given the filename as input.
+"""
+function load_from_polymake(::Type{SubdivisionOfPoints}, jsondict::Dict{Symbol, Any})
+    bigobject = Polymake.call_function(:common, :deserialize_json_string, json(jsondict))
+    typename = Polymake.type_name(bigobject)
+    if typename != "SubdivisionOfPoints<Rational>"
+        throw(ArgumentError("Loaded object is not of polymake type SubdivisionOfPoints, it has type " * typename))
+    end
+    return SubdivisionOfPoints(bigobject)
+end
+
+
+
+##############################################################################
+
+"""
+    load_from_polymake(::Type{PolyhedralComplex}, jsondict::Dict{Symbol, Any})
+
+Load a polyhedral complex stored in JSON format, given the filename as input.
+"""
+function load_from_polymake(::Type{PolyhedralComplex}, jsondict::Dict{Symbol, Any})
+    bigobject = Polymake.call_function(:common, :deserialize_json_string, json(jsondict))
+    typename = Polymake.type_name(bigobject)
+    if typename != "PolyhedralComplex<Rational>"
+        throw(ArgumentError("Loaded object is not of polymake type PolyhedralComplex, it has type " * typename))
+    end
+    return PolyhedralComplex(bigobject)
+end
+
+
+
+
+
+
+# Distinguish between the various polymake datatypes.
+function load_from_polymake(jsondict::Dict{Symbol, Any})
+    typename = jsondict[:_type]
+    if typename == "polytope::Cone<Rational>"
+        return load_from_polymake(Cone, jsondict)
+    elseif typename == "polytope::Polytope<Rational>"
+        if haskey(jsondict, "is_OSCAR_LP")
+            load_from_polymake(LinearProgram, jsondict)
+        else
+            load_from_polymake(Polyhedron, jsondict)
+        end
+    elseif typename == "fan::PolyhedralFan<Rational>"
+        return load_from_polymake(PolyhedralFan, jsondict)
+    elseif typename == "fan::PolyhedralComplex<Rational>"
+        return load_from_polymake(PolyhedralComplex, jsondict)
+    elseif typename == "fan::SubdivisionOfPoints<Rational>"
+        return load_from_polymake(SubdivisionOfPoints, jsondict)
+    elseif typename == "topaz::SimplicialComplex"
+        return load_from_polymake(SimplicialComplex, jsondict)
+    elseif typename == "common::GraphAdjacency<Undirected>"
+        return load_from_polymake(Graphs.Graph{Graphs.Undirected}, jsondict)
+    elseif typename == "common::GraphAdjacency<Directed>"
+        return load_from_polymake(Graphs.Graph{Graphs.Directed}, jsondict)
```

Comment:
Where is `load_from_polymake` for graphs defined? I didn't find it. Or is this a placeholder for a future extension?

Reply:
Incomplete refactor, sorry, I added this just now.